src/main/java/org/objectionary/parsing/Entities.java

```diff
@@ -88,11 +90,27 @@ public Entity one() {
     /**
      * Reads nested entity.
      * @return The parsed nested entity.
-     * @todo #49:30min Implement this method.
-     *  This method should read nested entity recursively.
      */
     public Map<String, Entity> nested() {
-        return new HashMap<>();
+        final Map<String, Entity> result = new HashMap<>();
+        while (true) {
+            final Token token = this.tokenizer.getToken();
+            if (token instanceof BracketToken) {
+                final BracketToken bracket = (BracketToken) token;
+                if (bracket.getState() == BracketToken.BracketType.CLOSE) {
+                    break;
+                }
+            }
+            assert token instanceof StringToken;
```

Comment:
@EugeneDar Can you get rid of assertion? https://www.yegor256.com/2016/06/17/dont-use-java-assertions.html